### DISCUSSION TITLE
hatch PyApp build

### DISCUSSION THREAD

**Discussion by sagregrevc** [ofek/pyapp#98] [Q&A]
@ofek I trued to use hatch with PyApp for building app, https://ofek.dev/pyapp/latest/how-to/
but "configuration" step hard understand. where it should be done?
it dont look like pyproject.toml or I dont know top key words

if I trued to use "cargo build --release", but it ask for cargo.toml and manifest, but in instruction nothing about this

if I trued with "hatch build --target app" then it will give .exe file with 3 Mb(probably 100 times less then expected) and 5 files inside which dont look like python
".data, .pdata .rdata .reloc .text"

additional question: py software running from IDE with command "streamlit run ./src/main.py --client.toolbarMode=minimal".
this request somethinh special from me if I use PyApp+hatch?

**Comment by ofek** [comment]
I think you're going for this example: https://ofek.dev/pyapp/latest/examples/#offline-installation

Before archiving the distribution you should install your required dependencies within.

**Reply by sagregrevc** [reply]
my main question: where I should use this options? 
my terminal can recognize hatch and cargo, but it and me dont have idea where or how use "export" from gif https://ofek.dev/pyapp/latest/

**Reply by ofek** [reply]
Those are environment variables that you have to set. The way to do that differs depending on what shell you are using.

**Reply by sagregrevc** [reply]
ok, started understand my mistake. 
but can it be done by hatch and pyproject.toml ?
Or this should be done only manually each time for new project?

do you have codes for win? it little bit hard to decode them. 
![image](https://github.com/ofek/pyapp/assets/115998253/e4cfdf41-b086-4896-9f0b-5f3e1f49e3b9)

what should exist before I use PyApp? only whl, gz ?

**Reply by ofek** [reply]
I'm sorry but I don't have time to help with support. Can you please be very specific with your questions?

I would ask on StackOverflow or one of the top AI chat bots from OpenAI or Anthropic.

**Reply by sagregrevc** [reply]
specific: what should exist before using PyApp?

**Reply by ofek** [reply]
- Your project built as a wheel
- One of these https://github.com/indygreg/python-build-standalone/releases with your dependencies already installed and then compressed into an archive again
